### PR TITLE
dev-bug - generate docs

### PR DESCRIPTION
**Pull request by MLainer1**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/CIAC-4133)

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Must have
- [ ] Tests
- [ ] Documentation
- [ ] [VS Code Extension](https://docs.gitlab.com/ee/ci/yaml/#tags)
  - [ ] Functionality implemented in the extension - <u>\<link-to-pr-here\></u>
  - [ ] N/A - Functionality cannot be implemented in the extension because <u>\<add-reason-here\></u>

### CODE REVIEW

File: demisto_sdk/commands/generate_docs/generate_integration_doc.py [hunk 1]
```diff
@@ -35,13 +35,15 @@ def append_or_replace_command_in_docs(old_docs: str, new_doc_section: str, comma
     Returns:
         str: The whole documentation.
     """
-    regexp = DOCS_COMMAND_SECTION_REGEX.format(command_name)
+    regexp = DOCS_COMMAND_SECTION_REGEX.format(command_name + "\n")
     # Read doc content
     errs = list()
-    if re.findall(regexp, old_docs, flags=re.DOTALL):
+    list_of_commands_to_replace = re.findall(regexp, old_docs, flags=re.DOTALL)
+    if list_of_commands_to_replace:
```
Comment (dorschw) [inline demisto_sdk/commands/generate_docs/generate_integration_doc.py:42]:
```suggestion
    if commands_to_replace := re.findall(regexp, old_docs, flags=re.DOTALL): 
```

File: demisto_sdk/commands/generate_docs/generate_integration_doc.py [hunk 2]
```diff
@@ -35,13 +35,15 @@ def append_or_replace_command_in_docs(old_docs: str, new_doc_section: str, comma
...
         new_docs = re.sub(regexp, new_doc_section, old_docs, flags=re.DOTALL)
         print_color('New command docs has been replaced in README.md.', LOG_COLORS.GREEN)
     else:
+
```
Comment (dorschw) [inline demisto_sdk/commands/generate_docs/generate_integration_doc.py:46]:
```suggestion
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,7 @@
 * Fixed an issue in the **download** command, where an exception would be raised when downloading system playbooks.
 * Fixed an issue where the **upload** failed on playbooks containing a value that starts with `=`.
 * Fixed an issue where the **generate-unit-tests** failed to generate assertions, and generate unit tests when command names does not match method name.
+*Fixed an issue where **generate-docs** changed the wrong command docs.
```
Comment (dantavori) [inline CHANGELOG.md:8]:
```suggestion
*Fixed an issue where running **generate-docs** with `-c` argument updated incorrectly other commands' sections.
```

File: demisto_sdk/commands/generate_docs/tests/generate_docs_test.py [hunk 4]
```diff
@@ -676,10 +676,13 @@ class TestAppendOrReplaceCommandInDocs:
     command = 'dxl-send-event'
     old_doc = open(positive_test_data_file).read()
     new_docs = "\n<NEW DOCS>\n"
+    new_command = '\n### dxl-send-event-new-one\n***\nSends the specified event to the DXL fabric.\n##### Base Command\n`dxl-send-event-new-one`'
```
Comment (dantavori) [inline demisto_sdk/commands/generate_docs/tests/generate_docs_test.py:679]: this change is unclear, can you explain?

File: CHANGELOG.md (hunk 3 above)
Comment (dorschw) [reply CHANGELOG.md:8]:
```suggestion
*Fixed an issue where **generate-docs** with `-c` argument updated sections of the incorrect commands.
```

File: demisto_sdk/commands/generate_docs/tests/generate_docs_test.py (hunk 4 above)
Comment (MLainer1) [reply demisto_sdk/commands/generate_docs/tests/generate_docs_test.py:679]: Added a new command in the .md file of the test

**Review by dorschw — APPROVED** [review]
🤡

**Pull request merged**